src/lambdaconnect_model/core.clj

```diff
@@ -44,20 +45,67 @@
   [path entities-by-name]
   (scoping/read-pull-scoping-edn path entities-by-name))
 
+(defn get-minim-scoping-sets 
+  "Given a validated scoping
+   returns map of sets which has tags as key and each set keeps
+   tags required for scoping that tag"
+  [validated-scoping]
+  (single/get-minimum-scoping-sets validated-scoping))
+
+(defn scope-selected-tags-with-tree
+  "Takes a snapshot, a user object from DB, entities-by-name, parsed (and validated) EDN of scoping rules, map of sets indiacting which tags must be scoped per tag and a set of desired tags.
+   It is advised to calculacte scoping sets once and pass the result.
+  A typical invocation looks like this: 
+  (scope-selected-tags-with-tree config (d/db db/conn) user entities-by-name validated-scope scoping-sets #{:RARestaurant.ofOwner})))
+
+  Returns a map with db ids, something like:
+  {:RAOwner.me #{11122, 1222} :user #{2312312}}
+  "
+  [config snapshot user entities-by-name scoping-defintion tag-scope tags]
+  (scoping/scope-selected-tags-with-tree config snapshot user entities-by-name scoping-defintion tag-scope tags))
+
+(defn scope-selected-tags
+  "Takes a snapshot, a user object from DB, entities-by-name, parsed (and validated) EDN of scoping rules, a set of desired tags and push?.
+  A typical invocation looks like this: 
+  (scope-selected-tags config (d/db db/conn) user entities-by-name validated-scope desired-tags #{:RARestaurant.ofOwner})))
```

Comment:
coś tu się nie zgadza - argumenty tej funkcji to: 
```[config snapshot user entities-by-name scoping-defintion tags push?]``` 
a podajesz w przykładowym wywołaniu:
```(scope-selected-tags config (d/db db/conn) user entities-by-name validated-scope desired-tags #{:RARestaurant.ofOwner})```
Wydaje mi się, że coś tutaj się nie zgadza, `push?` nie powinien być zbiorem tylko boolem.